File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.cpp
#include <vector>
#include <iostream>
#include <algorithm>
#include <math.h>
#include<numeric>
using namespace std;

int solution(vector<int> nums) {
    int answer = 0;
    	sort(nums.begin(), nums.end());
	int sum = accumulate(nums.begin(), nums.end(),0);
	vector<int> arr = { 2,3 };

	for (int i = 4; i <= sum; i++)
	{
		int chk = false;
		for (int j = 0; j < arr.size(); j++)
		{
			
			if ((int)sqrt(i) < arr[j]) break;

			if (i % arr[j] != 0) chk = true;
			else
			{
				chk = false;
				break;
			}
		}
		if (chk) arr.push_back(i);
	}


	vector<bool> mask(nums.size(), false);
	fill(mask.end() - 3, mask.end(), true);

	do{
		int sum = 0;
		for (int i = 0; i < nums.size(); i++)
		{
			if (mask[i]) sum += nums[i];
			
		}
		if (find(arr.begin(), arr.end(), sum) != arr.end())
			answer++;
	} while (next_permutation(mask.begin(), mask.end()));

    return answer;
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.cs
using System;
class Solution
{
    public int solution(int[] nums)
    {
        int answer = 0;
        int len = nums.Length;
        int all = (len*(len-1)*(len-2)) / (3*2);
        
        int idx1 = 0;
        int idx2 = 1;
        int idx3 = 2;
        
        for(int i = 0 ;i<all; i++)
        {
            int sum = nums[idx1]+ nums[idx2]+nums[idx3];
            
            if(Find(sum)) answer++;
            
            idx3++;
            if(idx3 >= len)
            {
                idx2++;
                idx3 = (idx2+1);
            }
            if(idx2 >= len-1)
            {
                idx1++;
                idx2 = idx1+1;
                idx3 = idx2+1;
            }            
            
        }

        return answer;
    }
    
    bool Find(int sum)
    {
        for(int i = (int)Math.Sqrt(sum); i>=2;i--)
        {
            if(sum % i == 0) return false;
        }
        return true;
    }
}